### PR TITLE
Fix DST timezone handling for viewed event weeks

### LINKED ISSUE #227: Showing wrong timezone when scheduling across a DST change
Image

I am trying to schedule an event for April.

I am in central Europe: my timezone _today_ (March 18th) is GMT+1, but in April it will be GMT+2. As you can see from the screenshot, it's wrongly using the GMT+1 timezone.

### PR DESCRIPTION
**Pull request by wslany**
Fixes #227

This PR is intentionally split into several commits:

1. [4ceda5a](https://github.com/schej-it/timeful.app/pull/228/commits/4ceda5acef066d20267b7853cfc6dd1991798c20) adds the DST regression proof, focused helper tests, and frontend CI.
2. [ece4641](https://github.com/schej-it/timeful.app/pull/228/commits/ece4641d7ac3a129a88e638647e3cb88214a48d6) applies the actual fix that makes the remaining regression pass.
3. [5b76640](https://github.com/schej-it/timeful.app/pull/228/commits/5b76640711f99ba49b6a01d737036d381ee6a58a) cleanup that removes a deprecated Vite CJS warning
4. [55dd935](https://github.com/schej-it/timeful.app/pull/228/commits/55dd935ff77b88b74868b79843adc345c72a5194) adds another regression test showing the correctness for fall-back DST changes, and another test for the non-edge case where the standard offset is kept during non-DST viewed weeks

I split it this way so review can follow a clear red/green sequence:
- first, prove the bug exists and that the regression test catches it,
- then, apply the smallest fix that turns the remaining failing regression green.

The first commit also includes unit tests for the new helper functions so the extracted date/offset logic is documented directly in tests, not only in implementation.

### CODE REVIEW

File: frontend/vitest.config.mjs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { defineConfig } from "vitest/config"
+import path from "path"
+
+export default defineConfig({
+  resolve: {
+    alias: {
+      "@": path.resolve(__dirname, "./src"),
```
Comment (maxwellward) [inline frontend/vitest.config.mjs]: Can we use `path.dirname(fileURLToPath(import.meta.url))` instead of `__dirname` here for better compatibility?
